`TestWorkflow_SearchArtifacts` failing on master

Caused by #8557.
https://github.com/argoproj/argo-workflows/actions/workflows/ci-build.yaml?query=branch%3Amaster

@rohankmr414 could you please take a look?

```
error: error pruning namespaced object /v1, Kind=PersistentVolumeClaim: rpc error: code = Unavailable desc = transport is closing
make: *** [Makefile:434: install] Error 1
```
@alexec I guess there is some issue with the installation. Not sure if #8557 caused this.

--- a/pkg/apis/workflow/v1alpha1/workflow_types_test.go
+++ b/pkg/apis/workflow/v1alpha1/workflow_types_test.go
@@ -825,28 +825,45 @@ func TestWorkflow_SearchArtifacts(t *testing.T) {
 
 	query := NewArtifactSearchQuery()
 
+	countArtifactName := func(ars ArtifactSearchResults, name string) int {
+		count := 0
+		for _, ar := range ars {
+			if ar.Artifact.Name == name {
+				count++
+			}
+		}
+		return count
+	}
+	countNodeID := func(ars ArtifactSearchResults, nodeID string) int {
+		count := 0
+		for _, ar := range ars {
+			if ar.NodeID == nodeID {
+				count++
+			}
+		}
+		return count
+	}
+
 	// no filters
 	queriedArtifactSearchResults := wf.SearchArtifacts(query)
 	assert.NotNil(t, queriedArtifactSearchResults)
 	assert.Len(t, queriedArtifactSearchResults, 3)
-	assert.Equal(t, "artifact-foo", queriedArtifactSearchResults[0].Artifact.Name)
-	assert.Equal(t, "artifact-bar", queriedArtifactSearchResults[1].Artifact.Name)
-	assert.Equal(t, "artifact-foobar", queriedArtifactSearchResults[2].Artifact.Name)
-	assert.Equal(t, "node-foo", queriedArtifactSearchResults[0].NodeID)
-	assert.Equal(t, "node-foo", queriedArtifactSearchResults[1].NodeID)
-	assert.Equal(t, "node-bar", queriedArtifactSearchResults[2].NodeID)
+	assert.Equal(t, 1, countArtifactName(queriedArtifactSearchResults, "artifact-foo"))
+	assert.Equal(t, 1, countArtifactName(queriedArtifactSearchResults, "artifact-bar"))
+	assert.Equal(t, 1, countArtifactName(queriedArtifactSearchResults, "artifact-foobar"))
+	assert.Equal(t, 2, countNodeID(queriedArtifactSearchResults, "node-foo"))
+	assert.Equal(t, 1, countNodeID(queriedArtifactSearchResults, "node-bar"))
 
 	// artifact GC strategy
 	query.ArtifactGCStrategies[ArtifactGCOnWorkflowCompletion] = true
 	queriedArtifactSearchResults = wf.SearchArtifacts(query)
 	assert.NotNil(t, queriedArtifactSearchResults)
 	assert.Len(t, queriedArtifactSearchResults, 3)
-	assert.Equal(t, "artifact-foo", queriedArtifactSearchResults[0].Artifact.Name)
-	assert.Equal(t, "artifact-bar", queriedArtifactSearchResults[1].Artifact.Name)
-	assert.Equal(t, "artifact-foobar", queriedArtifactSearchResults[2].Artifact.Name)
-	assert.Equal(t, "node-foo", queriedArtifactSearchResults[0].NodeID)
-	assert.Equal(t, "node-foo", queriedArtifactSearchResults[1].NodeID)
-	assert.Equal(t, "node-bar", queriedArtifactSearchResults[2].NodeID)
+	assert.Equal(t, 1, countArtifactName(queriedArtifactSearchResults, "artifact-foo"))
+	assert.Equal(t, 1, countArtifactName(queriedArtifactSearchResults, "artifact-bar"))
+	assert.Equal(t, 1, countArtifactName(queriedArtifactSearchResults, "artifact-foobar"))
+	assert.Equal(t, 2, countNodeID(queriedArtifactSearchResults, "node-foo"))
+	assert.Equal(t, 1, countNodeID(queriedArtifactSearchResults, "node-bar"))
 
 	// template name
 	query = NewArtifactSearchQuery()
@@ -872,10 +889,9 @@ func TestWorkflow_SearchArtifacts(t *testing.T) {
 	queriedArtifactSearchResults = wf.SearchArtifacts(query)
 	assert.NotNil(t, queriedArtifactSearchResults)
 	assert.Len(t, queriedArtifactSearchResults, 2)
-	assert.Equal(t, "artifact-foo", queriedArtifactSearchResults[0].Artifact.Name)
-	assert.Equal(t, "artifact-bar", queriedArtifactSearchResults[1].Artifact.Name)
-	assert.Equal(t, "node-foo", queriedArtifactSearchResults[0].NodeID)
-	assert.Equal(t, "node-foo", queriedArtifactSearchResults[1].NodeID)
+	assert.Equal(t, 1, countArtifactName(queriedArtifactSearchResults, "artifact-foo"))
+	assert.Equal(t, 1, countArtifactName(queriedArtifactSearchResults, "artifact-bar"))
+	assert.Equal(t, 2, countNodeID(queriedArtifactSearchResults, "node-foo"))
 
 	// bad query
 	query = NewArtifactSearchQuery()

